api/label/set.go

```diff
@@ -116,8 +123,10 @@ func (l *Set) Value(k kv.Key) (kv.Value, bool) {
 	if l == nil {
 		return kv.Value{}, false
 	}
-	rValue := l.equivalent.reflect()
-	vlen := rValue.Len()
+	var vlen int
+	var rValue reflect.Value
+	vlen = rValue.Len()
+	rValue = l.equivalent.reflect()
```

Comment:
Is the swapping of the initialization order here intentional?  Won't this initialize `vlen` to 0, resulting in `sort.Search` immediately returning 0 as it is searching on the range [0,0), which then causes the `if idx >= vlen` check to be the only case that can hit as `idx == vlen` will always be true?

Reply:
This was actually panicking! I've reversed those lines and added a test. Thanks! 😀